src/fft/fftn.rs

```diff
@@ -0,0 +1,534 @@
+use mlx_macros::default_device;
+
+use crate::{array::Array, error::FftError, stream::StreamOrDevice};
+
+use super::{
+    resolve_size_and_axis_unchecked, resolve_sizes_and_axes_unchecked, try_resolve_size_and_axis,
+    try_resolve_sizes_and_axes,
+};
+
+/// One dimensional discrete Fourier Transform.
+///
+/// # Params
+///
+/// - `a`: The input array.
+/// - `n`: Size of the transformed axis. The corresponding axis in the input is truncated or padded
+///   with zeros to match `n`. The default value is `a.shape[axis]`.
```

Comment:
```suggestion
///   with zeros to match `s`. The default value is `a.shape[axis]`.
```

Reply:
This follows the python api. 1D fft uses `n`, n-D uses `s`